### ISSUE TITLE
[Android Bootcamp] Twitter client

### ISSUE THREAD

**Issue by sophiataskova** [sophiataskova/SimpleTwitterClient#1]
/cc @codepathreview @codepath submitting HW #3 but still working on persistence and replying to tweets (hit a roadblock on using activeandroid, and just got ratelimited by twitter, so gif is also to come)

**Comment by nesquena** [comment]
> just got ratelimited by twitter

Yep, should be back within 15m: https://dev.twitter.com/docs/rate-limiting/1.1 so you can record the GIF

> hit a roadblock on using activeandroid

Anything I can help with? Check the [common questions](http://guides.codepath.com/android/ActiveAndroid-Guide#common-questions) and let me know if there's anything I can update in tips of guide.

Full review coming soon.

**Comment by codepathreview** [comment]
:+1: nice work. A few notes after checking out the code:
- Good to see you were able to format the timestamp cleanly
- Nice to see you used the `SwipeRefreshLayout` for easy timeline refreshes
- Good to see you refreshed the timeline after composing a new tweet
-  Consider adding the character count for Compose Activity to limit to 140. See [this event handler](http://guides.codepath.com/android/Basic-Event-Listeners#textchangedlistener) for more details.
- Glad to see you made the compose activity into a dialog instead
- Consider [better organizing your source files](http://guides.codepath.com/android/Organizing-your-Source-Files)
- Consider using the [ViewHolder pattern](https://github.com/codepath/android_guides/wiki/Using-an-ArrayAdapter-with-ListView#improving-performance-with-the-viewholder-pattern) to improve performance of tweets `ListView`.

Here's a detailed [Project 3 Feedback Guide here](http://courses.codepath.com/snippets/intro_to_android/project_3_feedback) which covers the most common issues with this submitted project. Read through the feedback guide point-by-point to determine how you could improve your submission. 

Let us know if you have any other thoughts or questions about this assignment. Hopefully by now you feel pretty comfortable with all the major pieces to basic Android apps (Views, Controllers, ActionBar, Navigation, Models, Authentication, API Communication, Persistence, et al) and see how they all fit together. We are close now to a turning point in the course where you should be hitting a "critical mass" towards your knowledge of Android.
